injection/clients/namespacedkube/informers/factory/factory.go

```diff
@@ -0,0 +1,53 @@
+/*
+Copyright 2020 The Knative Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package factory
+
+import (
+	context "context"
+
+	informers "k8s.io/client-go/informers"
+	client "knative.dev/pkg/client/injection/kube/client"
+	controller "knative.dev/pkg/controller"
+	injection "knative.dev/pkg/injection"
+	logging "knative.dev/pkg/logging"
+	"knative.dev/pkg/system"
+)
+
+func init() {
+	injection.Default.RegisterInformerFactory(withInformerFactory)
+}
+
+// Key is used as the key for associating information with a context.Context.
+type Key struct{}
+
+func withInformerFactory(ctx context.Context) context.Context {
+	c := client.Get(ctx)
+	return context.WithValue(ctx, Key{},
+		informers.NewSharedInformerFactoryWithOptions(c, controller.GetResyncPeriod(ctx),
+			// This factory scopes things to the system namespace.
+			informers.WithNamespace(system.Namespace())))
```

Comment:
I am super unthrilled to have this code depend on system.Namespace, but that should work :)
But also we can only use it for the system namespace, I can't limit it to any other (which we might want at some point, maybe).

Reply:
Honestly, we've been talking about restricting namespaces on controllers for ages, and I'm not aware of any good mechanism that exists for restricting informers to anything larger than a single namespace (e.g. namespace selector), and in that world I think you quickly end up with that namespace being the system namespace.

I'm happy to discuss other alternatives, but I like that this gets the job done for everything downstream (e.g. Tekton) without needing any changes except updating vendors.

I have some thoughts on how we might generalize this beyond `system.Namespace()` in a way that lives side-by-side with cluster-scoped informers (using more code-generation), but I'm just not convinced it's worth the extra effort right now.

Reply:
What I rather mean, is I'd prefer that the namespace is on the context somehow, rather than hardcoded.. but 🤷‍♂ 
/lgtm

Reply:
Captured this in an issue:

https://github.com/knative/pkg/issues/1159